.github/workflows/actions.yml

```diff
@@ -30,6 +30,9 @@ jobs:
           pip install -r requirements-build.txt
           pip install .
           pip install coveralls
+          pip install -r benchadapt/python/requirements.txt
+          pip install benchadapt/python
+          pip install benchrun/python
```

Comment:
I think the best way to handle this step is to put it all on one line. That way, if a package is specified in multiple places (e.g. a requirements.txt and in this command), pip should put a higher priority on the command-specified one. I also prefer to use `-e` for installing from source in CI, because I've found that omitting that flag can result in the wrong cached package being used in rare occasions. So the end result might look like this:
```
pip install \
  -e . \
  -e benchadapt/python \
  -e benchrun/python \
  -r requirements-test.txt \
  -r requirements-build.txt
```
This assumes you add `coveralls` to `requirements-test.txt`, which I think is a good idea. It's also redundant to specify the requirements files that are already specified in the `setup.py` files, like `requirements-cli.txt` or `benchadapt/python/requirements.txt`.

The only drawback of this approach is if there are conflicting dep versions then pip will fail, but almost none of our deps are version-tagged so I think this should work. It also closer mocks how `pip` will act when a user tries to install from pypi (in which case there'd definitely be a problem if we have conflicts).

Reply:
Oh this makes so much sense, thank you! And yeah, we can leave out the requirements files already picked up by setup.py; not sure why I didn't realize that. A while ago I changed the conbench `setup.py` to at least install its core requirements on pip install, so this that can get less bad now too. I forget if I did the thing so you can do `pip install conbench[dev]` or whatever; I don't think I did? Not sure if people normally do that for testing deps. Seems nice imo.

Reply:
😍 
```
        run: |
          pip install -e .[dev] \
            -e benchadapt/python \
            -e benchrun/python
```
I think it also runs a little faster this way because it doesn't have to do multiple rounds of dependency resolution